cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CruiseControlReconciler.java

```diff
@@ -247,52 +244,51 @@ protected Future<Void> certificatesSecret(Clock clock) {
-     * That way, each component is responsible for managing its credentials, and this method is responsible for keeping the Cruise Control's authentication file in sync.
+     * Generates a new centralized Cruise Control API secret containing an authentication file comprising of all
+     * REST API credentials, it is generated using the following API secrets:
+     *
+     *  (1) Previous Centralized API secret
+     *      The centralized API secret contains passwords for the healthcheck and rebalance operator users as well
+     *      the authentication file comprising of all REST API credentials.
+     *
+     *  (2) User-managed API secret
+     *      The user-managed API secret contains a custom authentication file and can specify any number of API users.
+     *
+     *  (3) Topic Operator-managed API secret
+     *      The Topic Operator-managed API secret contains a username and password for the Topic Operator user, which
+     *      has ADMIN role by default.
+     *
+     * If there are any changes to the API user content in any of the secrets, the centralized Cruise Control secret is
+     * updated with the new changes. Otherwise the content in the centralized Cruise Control secret remains the same.
      *
-     * If Cruise Control secret is present and there are no changes to the Topic Operator's secret content, we reuse the old Cruise Control secret's content.
-     * If Cruise Control secret is not present or there are changes in the Topic Operator's secret content, we generate a new Cruise Control secret, which includes the Topic Operator's credentials.
-     * 
-     * If the Topic Operator component is not enabled and Cruise Control's secret is present, we reuse the old Cruise Control secret's content. 
-     * If the Topic Operator component is not enabled and Cruise Control's secret is not present, we generate a new Cruise Control secret.
-     * 
-     * In any case, we generate the Cruise Control secret's content hash, that is later used to detect changes that require a pod restart.
+     * Strimzi maintains a hash of the centralized Cruise Control API secret content to detect changes that require a pod restart.
      *
      * @return Future which completes when the reconciliation is done.
      */
     protected Future<Void> apiSecret() {
         if (cruiseControl != null) {
-            if (isTopicOperatorEnabled) {
-                return Future.join(
-                    secretOperator.getAsync(reconciliation.namespace(), CruiseControlResources.apiSecretName(reconciliation.name())),
-                    secretOperator.getAsync(reconciliation.namespace(), KafkaResources.entityTopicOperatorCcApiSecretName(reconciliation.name()))
-                ).compose(
+            Future<Secret> centralizedApiSecretFuture = secretOperator.getAsync(reconciliation.namespace(), CruiseControlResources.apiSecretName(reconciliation.name()));
```

Comment:
What is `centralizedApiSecretFuture`? I don't think the `centralized` has a clear meaning. Maybe `ccApiSecretFuture`?